### DISCUSSION TITLE
[SOLVED] QueryBuilder IN Postgresql

### DISCUSSION THREAD

**Discussion by takealook-global** [phalcon/cphalcon#15289] [Q&A]
I have a int2(smallint) on Postgresql 12.5

So if i pass 1 the query run ok.

if i pass 1,2,3 or '1','2','3',....

Have some way to get this work with placeholder?

I have used all bind types and ANY, but I get the same error.

```
 $select->andWhere("r.s IN(:s:) AND env.s IN(:s:)", [
                's' => $options['s']
            ], ['s' => \Phalcon\Db\Column::BIND_SKIP]);
```

I get this error if i use 1,2,3 int.

> "message":"SQLSTATE[HY093]: Invalid parameter number: :status0","status":false,"code":"HY093"

and this error for string '1','2'..

`Invalid text representation: 7 ERROR:  invalid input syntax for type smallint:`

Without placeholder works.
`$select->andWhere("r.sIN(".join(",",[1,2]).") AND env.s IN(1,2)", `

**Comment by Jeckerson** [comment]
Hello

```
->andWhere('r.s IN ({s:array})', ['s' => $options['s']])
```

Also, note that for repeated parameters, you will need to have unique name

```
->andWhere('r.s IN ({s:array}) AND r.s2 IN ({s2:array})', ['s' => $options['s'], 's2' => $options['s']])
```

https://docs.phalcon.io/4.0/en/db-models#binding-parameters

**Reply by takealook-global** [reply]
Thanks, about the repeated im aware, I just missed the {field:array}.
and this post was made in a wrong account(company)

**Reply by renatogabrielbr** [reply]
Thank you, the right account :)

**Reply by renatogabrielbr** [reply]
> Hello
> 
> ```
> ->andWhere('r.s IN ({s:array})', ['s' => $options['s']])
> ```
> 
> Also, note that for repeated parameters, you will need to have unique name
> 
> ```
> ->andWhere('r.s IN ({s:array}) AND r.s2 IN ({s2:array})', ['s' => $options['s'], 's2' => $options['s']])
> ```
> 
> https://docs.phalcon.io/4.0/en/db-models#binding-parameters

As we talk about Database, can we answer that?

https://github.com/phalcon/phalcon/discussions/99
